src/state-manager/TransactionConsensus.ts

```diff
@@ -1024,6 +1094,565 @@ class TransactionConsenus {
         }
       }
     )
+
+    Comms.registerGossipHandler(
+      'poqo-receipt-gossip',
+      (payload: AppliedReceipt2) => {
+        profilerInstance.scopedProfileSectionStart('poqo-receipt-gossip')
+        try {
+          const queueEntry = this.stateManager.transactionQueue.getQueueEntrySafe(payload.txid)
+          if (queueEntry == null) {
+            /* prettier-ignore */ if (logFlags.error) this.mainLogger.error(`poqo-receipt-gossip no queue entry for ${payload.txid}`)
+            return
+          }
+          if (queueEntry.hasSentFinalReceipt === true) {
```

Comment:
was this condition have any impact on gossip ? Gossip design assumed only 1 iteration of sending or there can be multiple, so this condition will create possible gaps?

Reply:
this could result in some skipped gossip if it happens enough.  I created new counter, and put this in a spreadsheet of error counters we need to monitor in tests or live networks.

Reply:
sys-368